sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -188,6 +189,62 @@ def _microsoft_teams_user_raw_id(identifier: MicrosoftTeamsUserIdentifier) -> st
     return '8:orgid:{}'.format(user_id)
 
 
+MicrosoftBotProperties = TypedDict(
+    'MicrosoftBotProperties',
+    bot_id=str,
+    is_global=bool,
+    cloud=Union[CommunicationCloudEnvironment, str]
+)
+
+
+class MicrosoftBotIdentifier(object):
```

Comment:
should we override the equality check on the raw id?  I see only for unknown identifier and CommunicationUserIdentifier it was overridden, though it checks for all properties, @petrsvihlik for python it was agreed in this way?

Reply:
I'm unaware about any convention on this.
Waiting for @petrsvihlik to comment.

Reply:
Honestly, I'd unify the behavior to be based solely on the RawId (similarly to net, java) because that's what is ultimately defining whether two identifiers are equal no matter how the object was initialized.

this might require 
1) replicating the following logic for CommunicationUserIdentifier
```
if self.raw_id is None:
            self.raw_id = _phone_number_raw_id(self)
```
2) adding `__eq__` where it's missing (phonenumber, msteams user/bot identifiers)

Reply:
the logic still checks for all the object attributes
`self.__dict__ == other.__dict__`, though it should check equality of the raw id. I am marking as unresolved again, pls let me know if I missing smth and would be great if we also cover the equality check logic with tests

Reply:
I get it now. Updated and added tests